cloudsync_onedrive.py

```diff
@@ -381,6 +386,16 @@ def _direct_api(self, action, path=None, *, url=None, stream=None, data=None, he
 
         return res
 
+    def _safe_direct_api(self, path, method="get", default=None):
```

Comment:
In all four places where this new method is used, the value coming back from `_safe_direct_api` is cached in a cache somewhere.  There doesn't seem to be any attention paid to the caches that now are empty, even though there is data that should be in them. A simple `CloudTemporaryError` can trigger this condition.

Reply:
LGTM